torch/csrc/autograd/engine.h

```diff
@@ -195,17 +200,22 @@ struct TORCH_API Engine {
 
   bool is_checkpoint_valid();
 
-  size_t ready_queue_size(at::Device device);
+  size_t ready_queue_size(const std::shared_ptr<GraphTask>& graph_task, at::Device device);
 
  protected:
   void compute_dependencies(Node* root, GraphTask& task);
   void evaluate_function(
       std::shared_ptr<GraphTask>& graph_task,
       Node* func,
       InputBuffer& inputs);
-  ReadyQueue& ready_queue(at::Device device);
-  ReadyQueue& ready_queue_by_index(int device_index);
-  void start_threads();
+  ReadyQueue& ready_queue(
+      const std::shared_ptr<GraphTask>& graph_task,
+      at::Device device);
+  ReadyQueue& ready_queue_by_index(
+      const std::shared_ptr<GraphTask>& graph_task,
+      int device_index);
+  // start device threads in Engine,
+  void start_device_threads();
```

Comment:
Thanks for the rename, it's much appreciated. (Maybe add, "does NOT start cpu thread"?)